File: msg-sentiment/src/test/java/dev/number6/sentiment/dagger/FakeCommonModule.java
package dev.number6.sentiment.dagger;

import dev.number6.comprehend.port.ComprehensionPort;
import dev.number6.db.port.DatabasePort;
import dev.number6.sentiment.adaptor.FakeComprehensionAdaptor;
import dev.number6.sentiment.adaptor.FakeDatabaseAdaptor;
import dagger.Module;
import dagger.Provides;

@Module
public class FakeCommonModule {

    @Provides
    public DatabasePort providesDatabasePort() {
        return new FakeDatabaseAdaptor();
    }

    @Provides
    public ComprehensionPort providesComprehensionPort() {
        return new FakeComprehensionAdaptor();
    }
}

File: msg-sentiment/src/test/kotlin/dev/number6/sentiment/dagger/FakeCommonModule.kt
package dev.number6.sentiment.dagger

import dagger.Module
import dagger.Provides
import dev.number6.comprehend.port.ComprehensionPort
import dev.number6.db.port.DatabasePort
import dev.number6.sentiment.adaptor.FakeComprehensionAdaptor
import dev.number6.sentiment.adaptor.FakeDatabaseAdaptor

@Module
class FakeCommonModule {
    @Provides
    fun providesDatabasePort(): DatabasePort {
        return FakeDatabaseAdaptor()
    }

    @Provides
    fun providesComprehensionPort(): ComprehensionPort {
        return FakeComprehensionAdaptor()
    }
}